### DISCUSSION TITLE
How to hide the menu such as "FILE" "ADD"?

### DISCUSSION THREAD

**Discussion by tao-bioinfo** [GMOD/jbrowse-components#3542] [Q&A] [answered]
We built a JBrowse page that can be visited worldwide, however we do not want to show the menu such as "FILE" "ADD"

What should I do? Thank you!

![image](https://user-images.githubusercontent.com/1146090/221064330-43f9eef5-876d-44d0-b6ab-1c1016dce087.jpeg)

**Comment by cmdcolin** [comment, marked as answer]
Currently there is no way to do this easily, but you could write a plugin that calls the "setMenus" function to be empty or just removes the File menu on the rootModel

See https://github.com/GMOD/jbrowse-components/blob/0cdf3432bfd6f28a361544374f5a713be4819db3/products/jbrowse-web/src/rootModel.ts#L637-L639

**Reply by MorganHis** [reply]
I'd like to know if hiding the menu bar can be made a configurable feature in a future version, thank you very much

**Reply by MorganHis** [reply]
your any suggestions will be highly appreciated！

**Reply by adampowell1astrazeneca** [reply]
@cmdcolin 

I noticed this is done on the Vite demo app: https://jbrowse.org/demos/lgv-vite/

How is it done there? I can't seem to work it out. It just seems to be using the npm version of jbrowse.

**Reply by cmdcolin** [reply]
@adampowell1astrazeneca that is a demo of the @jbrowse/react-linear-genome-view component, so it is just a single 'linear genome view' component that we turned into a npm installable package. it doesn't have any "top level menubar" with File/Add/etc. but it does have a "hamburger menu" which I refer to as the "view menu".

screenshot of the view menu on @jbrowse/react-linear-genome-view
![image](https://github.com/GMOD/jbrowse-components/assets/6511937/0cf32644-226e-4c86-8578-02796106ebf1)


there is also https://jbrowse.org/demos/app-vite/ which is a npm version of the "full jbrowse web app" that does have the top level menu with File/Add/etc.

let me know if i can advise further, here are some links though

https://jbrowse.org/jb2/docs/embedded_components/
https://jbrowse.org/storybook/lgv/main/?path=/docs/getting-started--docs
https://www.npmjs.com/package/@jbrowse/react-linear-genome-view

**Reply by cmdcolin** [reply]
didn't mention it above but there is also the version of jbrowse that you get from e.g. our quickstart guide with the "jbrowse create" command. that gives you just a pre-built zip file, not an npm package.

**Reply by adampowell1astrazeneca** [reply]
That's great. Thanks. 

But yes, currently using the `jbrowse create` version and was looking how best to configure it. Looks like switching to `@jbrowse/react-linear-genome-view` is the way to do it!

**Comment by MorganHis** [comment]
Has your problem been solved? I have the same needs and haven't figured out how to implement it. I use it in vue and want to hide the 'FILE' and 'ADD' buttons in the menu. It would be appreciated if you can provide solutions. gratitude.

**Reply by cmdcolin** [reply]
@MorganHis just catching up with this as I saw your new issue report. if you have a particular behavior you want to see with the menu (e.g. hide completely) can you make a new github issue to describe? thanks
